File: javascript/fibonacci.js
// program to check if number is fibonacci

/**
 * isPerfectSquare - checks if number is a perfect square
 * @n - a number of type int
 * 
 * return - true if n is perfect square else false
 */
function isPerfectSquare(n){
    let value = parseInt(Math.sqrt(n));
    return (value * value === n);
}

/**
 * isFibonacci - checks if a number is Fibonacci
 * @n - a number of type int
 * 
 * return - true if n is Fibonacci else false
 */
function isFibonacci(n){

    // n is Fibonacci if one of 5*n*n - 4 or both
    // is a perfect square
    return (isPerfectSquare(5*n*n + 4) ||
            isPerfectSquare(5*n*n - 4));
}

for (let i = 0; i <= 30; i++){
    isFibonacci(i)? console.log(i + " is a Fibonacci number.") :
    console.log(i + " is not a Fibonacci number.");
}

File: python/fibonacci.py
"""
confirm if a number belongs to the fibonacci sequence.
"""
# import of a math module
import math

def is_perfect_square(n):
    """ a function that returns true if n
    is a perfect square"""

    value = int(math.sqrt(n))
    return (value * value == n)

def is_fibonacci(n):
    """ checks if n is a fibonacci number"""

    # n is Fibonacci if one of 5*n*n + 4 or 5*n*n - 4 or both
    # is a perfect square.
    return (is_perfect_square(5*n*n + 4) or
                is_perfect_square(5*n*n - 4))

number_input = int(input("Enter a number: "))
if (is_fibonacci(number_input)== True):
    print("The number {} is fibonacci".format(number_input))
else:
    print("The number {} is not fibonacci".format(number_input))
